platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/VersionCatalogDependencyRegistry.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.buildinit.plugins.internal;
+
+import org.gradle.util.internal.TextUtil;
+
+import java.util.Collection;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class VersionCatalogDependencyRegistry {
+    private static final Pattern RESERVED_LIBRARY_PREFIX = Pattern.compile("^(bundles|versions|plugins)[- ]");
```

Comment:
🤔 Should these reference the constants in `TomlCatalogFileParser`?  Even better, should that type make available some `getReservedPrefixes/Components()` or `isResreved(String)` methods that could be used here rather than defining these in both the init and the VC code?

Reply:
The `TomlCatalogFileParser` doesn't actually validate the restrictions `DefaultVersionCatalogBuilder` does and also has the needed constants - but of course not accessible.

I made the constants accessible and use them to build the `Pattern` objects here, the code in the init plugin generates working aliases from potentially invalid input which means that validation would not help it.